compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala

```diff
@@ -181,49 +256,66 @@ class ScenarioService(
           interpret(context, packageId, scenarioId.getName)
             .map(_.map {
               case error: ScenarioRunner.ScenarioError =>
-                RunScenarioResponse.newBuilder
-                  .setError(
-                    new Conversions(
-                      context.homePackageId,
-                      error.ledger,
-                      error.currentSubmission.map(_.ptx),
-                      error.traceLog,
-                      error.warningLog,
-                      error.currentSubmission.flatMap(_.commitLocation),
-                      error.stackTrace,
-                    )
-                      .convertScenarioError(error.error)
+                Left(
+                  new Conversions(
+                    context.homePackageId,
+                    error.ledger,
+                    error.currentSubmission.map(_.ptx),
+                    error.traceLog,
+                    error.warningLog,
+                    error.currentSubmission.flatMap(_.commitLocation),
+                    error.stackTrace,
                   )
-                  .build
+                    .convertScenarioError(error.error)
+                )
               case success: ScenarioRunner.ScenarioSuccess =>
-                RunScenarioResponse.newBuilder
-                  .setResult(
-                    new Conversions(
-                      context.homePackageId,
-                      success.ledger,
-                      None,
-                      success.traceLog,
-                      success.warningLog,
-                      None,
-                      ImmArray.Empty,
-                    )
-                      .convertScenarioResult(success.resultValue)
+                Right(
+                  new Conversions(
+                    context.homePackageId,
+                    success.ledger,
+                    None,
+                    success.traceLog,
+                    success.warningLog,
+                    None,
+                    ImmArray.Empty,
                   )
-                  .build
+                    .convertScenarioResult(success.resultValue)
+                )
             })
         }
-        .map(_.flatMap(x => x))
+        .map(_.flatten)
+
+    Future {
+      val startedAt = Instant.now.toEpochMilli
+
+      var millisPassed: Long = 0
+      def sleepRandom() = {
+        val delta: Long = 300 + (Random.nextDouble() * 300).toLong
```

Comment:
just curious, why did you choose a random sleep delta instead of, say, just sleeping 0.5s each time?

Reply:
I did it so that there's a clearer visual update that's easier to track in the corner of your eye. I think 500ms would be hard to see most of the time, since only one or two digits would change, and at regular intervals.

Reply:
oh that's clever, I like it. Could you please add this as a comment so we don't forget?